### PR TITLE
Building IBP images on solar 'master' instead of downloading them

### PR DESCRIPTION
**Pull request by gitfred**
Enables IBP images building process on `vagrant up` command.

### CODE REVIEW

**Comment by gitfred** [comment]
@rustyrobot @prmtl please review

**Comment by loles** [comment]
looks good for me

File: examples/provisioning/provision.py [hunk 1]
```diff
@@ -42,17 +43,30 @@ def partitioning(self):
 # Dnsmasq resources
 for node in nodes_list:
     node = NodeAdapter(node)
-    node_resource = filter(lambda n: n.name.endswith('node_{0}'.format(node.node_id)), node_resources)[0]
+    node_resource = next(n for n in node_resources
+                         if n.name.endswith('node_{0}'.format(node.node_id)))
 
-    node_resource.update({'partitioning': node.partitioning})
+    with open('/vagrant/tmp/keys/ssh_public') as fp:
```
Comment (prmtl) [inline examples/provisioning/provision.py]: You can open it just once.

Comment (gitfred) [inline examples/provisioning/provision.py]: done

**Comment by prmtl** [comment]
:shipit:

**Pull request merged**